core/src/banking_stage/consumer.rs

```diff
@@ -206,6 +206,26 @@ impl Consumer {
             .slot_metrics_tracker
             .increment_retryable_packets_count(retryable_transaction_indexes.len() as u64);
 
+        // Now we track the performance for the interested transactions which is not in the retryable_transaction_indexes
+        // We assume the retryable_transaction_indexes is already sorted.
+        for (index, packet) in packets_to_process.iter().enumerate() {
+            if packet.original_packet().meta().is_perf_track_packet() {
+                if let Some(start_time) = packet.start_time() {
+                    if retryable_transaction_indexes.binary_search(&index).is_err() {
+                        let duration = Instant::now() - *start_time;
+
+                        debug!(
+                            "Banking stage processing took {duration:?} for transaction {:?}",
+                            packet.transaction().get_signatures().first()
+                        );
+                        inc_new_counter_info!(
```

Comment:
I thought about it. Doing accumulation reduce metrics cost. But it will lose some information like the max, min etc (we got a mean during the report period). And my other reasoning of this whole random mask matching's purpose is to make the this metric report rare enough as we only catch about 1/4096 of txns.